cmd/controller/controllerspc.go

```diff
@@ -0,0 +1,394 @@
+	//"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	apis "github.com/openebs/node-disk-manager/pkg/apis/ndm/v1alpha1"
+	clientset "github.com/openebs/node-disk-manager/pkg/client/clientset/versioned"
+	crdscheme "github.com/openebs/node-disk-manager/pkg/client/clientset/versioned/scheme"
+	informers "github.com/openebs/node-disk-manager/pkg/client/informers/externalversions"
+	listers "github.com/openebs/node-disk-manager/pkg/client/listers/ndm/v1alpha1"
+	//samplev1alpha1 "github.com/testsamplecontroller/sample-controller/temp"
+)
+
+const controllerAgentName = "node-disk-manager"
+
+const (
+	// SuccessSynced is used as part of the Event 'reason' when a spc is synced
+	SuccessSynced = "Synced"
+	// ErrResourceExists is used as part of the Event 'reason' when a spc fails
+	// to sync due to a spc of the same name already existing.
+	ErrResourceExists = "ErrResourceExists"
+
+	// MessageResourceExists is the message used for Events when a resource
+	// fails to sync due to a spc already existing
+	MessageResourceExists = "Resource %q already exists and is not managed by spc"
+	// MessageResourceSynced is the message used for an Event fired when a spc
+	// is synced successfully
+	MessageResourceSynced = "SPC synced successfully"
+)
+
+// Controller is the controller implementation for spc resources
+type Controller struct {
+	// kubeclientset is a standard kubernetes clientset
+	kubeclientset kubernetes.Interface
+	// sampleclientset is a clientset for our own API group
+	clientset clientset.Interface
+
+	spcLister listers.StoragePoolClaimLister
+
+	spcSynced cache.InformerSynced
+
+	// workqueue is a rate limited work queue. This is used to queue work to be
+	// processed instead of performing it as soon as a change happens. This
+	// means we can ensure we only process a fixed amount of resources at a
+	// time, and makes it easy to ensure we are never processing the same item
+	// simultaneously in two different workers.
+	workqueue workqueue.RateLimitingInterface
+	// recorder is an event recorder for recording Event resources to the
+	// Kubernetes API.
+	recorder record.EventRecorder
+}
+
+type QueueLoad struct {
+	key       string
+	operation string
+}
+
+// NewController returns a new sample controller
+func NewController(
+	kubeclientset kubernetes.Interface,
+	clientset clientset.Interface,
+	kubeInformerFactory kubeinformers.SharedInformerFactory,
+	spcInformerFactory informers.SharedInformerFactory) *Controller {
+
+	// obtain references to shared index informers for the sp and spc
+	// types.
+	spcInformer := spcInformerFactory.Ndm().V1alpha1().StoragePoolClaims()
```

Comment:
The entire controller is tied to this version. It will be good to think about versioning now itself. Perhaps name of this file along with the name of structures can be suffixed with versions. 
controller/v1alpha1 /spc.go or controller/spcv1alpha1.go or something else.